### PR TITLE
Redirect statistics finder to Research and Statistics finder

### PR DESCRIPTION
**Pull request by bilbof**
This will redirect two finders (stats and stats announcements) to the new Research and Statistics finder. The params will be preserved. This will only impact the English locale. At a later date we'll remove all of this code, once we have the translations for the new finders.

Trello: https://trello.com/c/jMeefvQ1/543

These two URLS:

- https://www.gov.uk/government/statistics
- https://www.gov.uk/government/statistics/announcements

Will redirect to:

- https://www.gov.uk/search/research-and-statistics

https://trello.com/c/BE2ooOcY/934-redirect-whitehall-statistics-finder-to-finder-frontend

### CODE REVIEW

**Comment by sihugh** [comment]
Has this been signed off by el gran queso?

**Comment by bilbof** [comment]
@sihugh, I'm still awaiting sign off from les grand fromage.

**Comment by sihugh** [comment]
He says hold for now

File: app/controllers/statistics_announcements_controller.rb [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
   enable_request_formats(index: [:js])
 
   def index
-    @filter = Frontend::StatisticsAnnouncementsFilter.new(filter_params)
-    expire_cache_for_index_on_next_announcement_expiry(@filter.results)
-    if request.xhr?
-      skip_slimmer
-      render partial: "statistics_announcements/filter_results"
-    end
+    redirect_to_research_and_statistics
```
Comment (barrucadu) [inline app/controllers/statistics_announcements_controller.rb]: Does this need a locale restriction?

Comment (bilbof) [reply app/controllers/statistics_announcements_controller.rb]: In this case it's not required, as there aren't other locales for statistics announcements. E.g. https://www.gov.uk/government/statistics/announcements.fr

File: app/controllers/statistics_controller.rb [hunk 2]
```diff
@@ -34,6 +38,24 @@ def index
...
+  def redirect_to_research_and_statistics(format = "")
+    base_path = "#{Plek.new.website_root}/search/research-and-statistics#{format}"
+    redirect_to("#{base_path}?#{research_and_statistics_query_string}")
+  end
+
+  def research_and_statistics_query_string
+    {
+      content_store_document_type: 'published_statistics',
+      keywords: params['keywords'],
+      level_one_taxon: params['taxons'].try(:first),
```
Comment (barrucadu) [inline app/controllers/statistics_controller.rb]: Should the original parameter be `taxons` or `topics`?  It's hard remembering which is used where...

Comment (bilbof) [reply app/controllers/statistics_controller.rb]: It's both!

Announcements uses topics: https://www.gov.uk/government/statistics/announcements?utf8=%E2%9C%93&topics%5B%5D=495afdb6-47be-4df1-8b38-91c8adb1eefc

Published stats uses taxons: https://www.gov.uk/government/statistics?keywords=&taxons%5B%5D=495afdb6-47be-4df1-8b38-91c8adb1eefc&departments%5B%5D=all&from_date=&to_date=

This code is on integration at the moment. If you go to either of the URLs above (on integration) you should be redirected correctly.

Comment (bilbof) [reply app/controllers/statistics_controller.rb]: Likewise, stats announcements uses "organisations", while stats uses "departments".

**Comment by bilbof** [comment]
Thanks, @barrucadu, I've pushed a fix for this. It's on integration now.

File: app/controllers/statistics_announcements_controller.rb [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
 
-  def expire_cache_for_index_on_next_announcement_expiry(announcements)
-    time_to_releases = announcements.map { |ann| ann.release_date - Time.zone.now }.reject { |time_span| time_span <= 0 }
-    expires_in((time_to_releases << Whitehall.default_cache_max_age).min)
+  def research_and_statistics_query_string
+    {
+      content_store_document_type: 'upcoming_statistics',
+      keywords: params['keywords'],
+      level_one_taxon: params['topics'].try(:first),
+      organisations: params['organisations'],
```
Comment (barrucadu) [inline app/controllers/statistics_announcements_controller.rb]: I think the "all" change will need making here too

Comment (bilbof) [reply app/controllers/statistics_announcements_controller.rb]: Thanks, I've changed this.

**Pull request merged**